controller/aks-cluster-config-handler.go

```diff
@@ -704,14 +704,13 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, secretsCache w
 			updateNodePool := false
 			upstreamNodePool, ok := upstreamNodePools[npName]
 			if ok {
-				if to.Bool(np.EnableAutoScaling) {
+				if to.Bool(np.EnableAutoScaling) && !to.Bool(upstreamNodePool.EnableAutoScaling) {
 					if to.Int32(np.Count) != to.Int32(upstreamNodePool.Count) {
 						return config, fmt.Errorf("can't update node count when autoscaling is enabled in node pool [%s] for cluster [%s]", to.String(np.Name), config.Spec.ClusterName)
 					}
```

Comment:
I think this should be checked in the block below as well. Another option is to refactor the logic so count doesn't need to be checked twice.

Reply:
If I'm understanding the bug correctly (which I'm not sure I am) that would defeat the purpose, the point is to check that both EnableAutoScaling and Count aren't changing at the same time. But I agree that refactoring to avoid this quadrupal-nested conditional would be nice if possible.

Reply:
This was already refactored and reflects all cases which we have to check with Azure API

Reply:
I thought that the point was, count cannot be changed if auto scaling is enabled.